File: src/services/get_available_restaurants.rs
use crate::{
  api::available_restaurants::{make_request, parse_response},
  models::Restaurant,
};
use reqwest::Error;

pub async fn get_available_restaurants(identifier: String) -> Result<Vec<Restaurant>, Error> {
  let response = make_request(identifier).await?;
  Ok(parse_response(response))
}

File: src/services/getAvailableRestaurants.ts
import { defaultFetcher, type Fetcher } from "@literate.ink/utilities";
import { buildRequest, parseResponse } from "~/api/availableRestaurants";
import type { Restaurant } from "~/models/Restaurant";

export async function getAvailableRestaurants(identifier: string, fetcher: Fetcher = defaultFetcher): Promise<Array<Restaurant>> {
  const response = await fetcher(buildRequest(identifier));
  return parseResponse(response);
}
